pkg/ksym/ksym.go

```diff
@@ -32,26 +33,41 @@ import (
 
 var ErrFunctionNotFound = errors.New("kernel function not found")
 
+type CacheStats struct {
```

Comment:
Rather than having this, I would prefer to have a metrics struct with the counters defined below.

Reply:
Are you referring to the counters created in `profiler.go` or something else? If so, I thought it would be a good separation of concerns to keep the stats here and create the Prometheus counters on the profiler pkg, but no strong opinions!

Reply:
What I meant was to have `metrics` struct in kysm package similar to one in profiler.go. And rather than having an additional struct for stats using counters directly in here. So we can pass a register to this type and handle everything separately. That's actually a pattern we use too often.